packages/commonwealth/scripts/decode-db-content.ts

```diff
@@ -280,12 +280,93 @@ async function decodeComments(lastId: number = 0) {
   }
 }
 
+async function decodeProfileBios(lastId: number = 0) {
+  let lastProfileId = lastId;
+  while (true) {
+    const transaction = await models.sequelize.transaction();
+    try {
+      const profiles = await models.sequelize.query<{
+        id: number;
+        profile_bio: string;
+      }>(
+        `
+            SELECT id, profile -> 'bio' as profile_bio
+            FROM "Users"
+            WHERE profile -> 'bio' != 'null'
+              AND profile -> 'bio' != '""'
+              AND id > :lastId
+            ORDER BY id
+            LIMIT :batch_size FOR UPDATE;
+        `,
+        {
+          transaction,
+          type: QueryTypes.SELECT,
+          replacements: {
+            lastId: lastProfileId,
+            batch_size: BATCH_SIZE,
+          },
+        },
+      );
+
+      if (profiles.length === 0) {
+        await transaction.rollback();
+        break;
+      }
```

Comment:
If I'm understanding correctly, this condition indicates that it's beyond the last page and/or there are no more items to process so it breaks out of the infinite loop.

Why does a rollback need to occur under that condition?

Reply:
It could be a rollback or a commit doesn't make a difference. The goal is just to close out the transaction.